cloudferrylib/os/actions/networks_transporter.py

```diff
@@ -13,16 +13,73 @@
 # limitations under the License.
 
 
+from fabric.api import settings
+
+from cloudferrylib.base.action import action
 from cloudferrylib.base.action import transporter
-from cloudferrylib.utils import utils as utl
+from cloudferrylib.utils import utils
+from cloudferrylib.utils import remote_runner
+
+
+NAMESPACE = 'routers'
 
 
 class NetworkTransporter(transporter.Transporter):
 
     def run(self, **kwargs):
-        src_resource = self.src_cloud.resources[utl.NETWORK_RESOURCE]
-        dst_resource = self.dst_cloud.resources[utl.NETWORK_RESOURCE]
+        src_resource = self.src_cloud.resources[utils.NETWORK_RESOURCE]
+        dst_resource = self.dst_cloud.resources[utils.NETWORK_RESOURCE]
         info = src_resource.read_info()
         dst_resource.deploy(info)
         return {}
 
+
+class ChangeRouterBase(action.Action):
+    def __init__(self, *args, **kwargs):
+        super(ChangeRouterBase, self).__init__(*args, **kwargs)
+        self.net = self.cloud.resources[utils.NETWORK_RESOURCE]
+        self.cfg = self.cloud.cloud_config.cloud
+
+    def modify_router(self, router):
+        node = self.net.get_router_host(router['id'])
+        runner = remote_runner.RemoteRunner(node, self.cfg.ssh_user,
+                                            password=self.cfg.ssh_sudo_password)
+        with settings(abort_on_prompts=True,
```

Comment:
[abort-on-prompts](http://docs.fabfile.org/en/1.10/usage/env.html#abort-on-prompts) just terminate command when it waits for user input. Commands here don't require any user input. I just copy-paste it from another place. Will delete it.